### PR TITLE
[BugFix] Fix thirdparty build

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This PR fix two issues.
+ Force use make build system, since ninja doesn't support install only headers.
+ Fix symbol multiple definition in ubuntu.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -995,8 +995,11 @@ build_jansson() {
     mkdir -p build
     cd build
     $CMAKE_CMD .. -DCMAKE_INSTALL_PREFIX=${TP_INSTALL_DIR} -DCMAKE_INSTALL_LIBDIR=lib
+    OLD_BUILD_SYSTEM=${BUILD_SYSTEM}
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:998]: no need to save to `$OLD_BUILD_SYSTEM`, directly use `make` in following two lines.

File: thirdparty/build-thirdparty.sh [hunk 2]
```diff
@@ -1006,9 +1009,12 @@ build_avro_c() {
     mkdir -p build
     cd build
     $CMAKE_CMD .. -DCMAKE_INSTALL_PREFIX=${TP_INSTALL_DIR} -DCMAKE_INSTALL_LIBDIR=lib64 -DCMAKE_BUILD_TYPE=Release
+    OLD_BUILD_SYSTEM=${BUILD_SYSTEM}
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:1012]: this is a cmake build, cmake will properly generate cmake or ninja conf, current is fine, no need to force a `make` build.

File: thirdparty/build-thirdparty.sh [hunk 3]
```diff
@@ -1026,12 +1032,14 @@ build_serdes() {
                 --LDFLAGS="-L ${TP_INSTALL_DIR}/lib -L ${TP_INSTALL_DIR}/lib64" \
                 --enable-static \
                 --disable-shared
-
+    OLD_BUILD_SYSTEM=${BUILD_SYSTEM}
+    BUILD_SYSTEM='make'
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:1036]: no need to save to `$OLD_BUILD_SYSTEM`, directly use `make` in following two lines.

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**